crates/websocket/src/routes.rs

```diff
@@ -57,6 +61,22 @@ impl Actor for WsSession {
     // before processing any other events.
     // across all routes within application
     let addr = ctx.address();
+
+    // Rate limit check a joining IP first, before doing any parsing
+    self
+      .rate_limiter
+      .message()
+      .check(self.ip.to_owned())
+      .into_actor(self)
+      .then(|res, act, ctx| {
+        if !res {
+          debug!("Websocket join with IP: {} has been rate limited.", act.ip);
+          ctx.stop()
+        }
+        actix::fut::ready(())
+      })
+      .wait(ctx);
```

Comment:
don't wait on the check future like this, `check` doesn't need to be a async

Reply:
you should also put rate limiting in the StreamHandler. Only limiting `connect` doesn't get you much, since connect is called once per session, and not on every message